histoqc/BaseImage.py

```diff
@@ -112,14 +141,36 @@ def getImgThumb(self, dim):
                             f"{self['filename']}: Desired Image Level {dim+1} does not exist! Instead using level {osh.level_count-1}! Downstream output may not be correct")
                         self["warnings"].append(
                             f"Desired Image Level {dim+1} does not exist! Instead using level {osh.level_count-1}! Downstream output may not be correct")
+                    if self["bounding_box"] == True:
+                        size=tuple((np.array(self["image_base_size"])/osh.level_downsamples[dim]).astype(int))
+                    else:
+                        size=osh.level_downsamples[dim]
                     logging.info(
                         f"{self['filename']} - \t\tloading image from level {dim} of size {osh.level_dimensions[dim]}")
-                    img = osh.read_region((0, 0), dim, osh.level_dimensions[dim])
-                    self[key] = np.asarray(img)[:, :, 0:3]
+                    tile = osh.read_region((xc, yc), dim, size)
+                    if np.shape(tile)[-1]==4:
+                        bg_color = '#' + osh.properties.get(openslide.PROPERTY_NAME_BACKGROUND_COLOR, 'ffffff')
+                        img = Image.new('RGB', size, bg_color)
+                        img.paste(tile, None, tile)
+                    else:
+                        img=tile
+                    img = np.asarray(img)
+                    self[key] = img
                 else:  # assume its an explicit size, *WARNING* this will likely cause different images to have different
                     # perceived magnifications!
                     logging.info(f"{self['filename']} - \t\tcreating image thumb of size {str(dim)}")
-                    self[key] = np.array(osh.get_thumbnail((dim, dim)))
+                    if self["bounding_box"] == True:
+                        downsample = max(*(dim / thumb for dim, thumb in zip(osh.dimensions, (dim, dim))))
+                        level = osh.get_best_level_for_downsample(downsample)
+                        size=tuple((np.array(self["image_base_size"])/osh.level_downsamples[level]).astype(int))
+                        tile = osh.read_region((xc, yc), level, size)
+                        bg_color = '#' + osh.properties.get(openslide.PROPERTY_NAME_BACKGROUND_COLOR, 'ffffff')
+                        thumb = Image.new('RGB', tile.size, bg_color)
+                        thumb.paste(tile, None, tile)
+                        thumb.thumbnail(size, getattr(Image, 'Resampling', Image).LANCZOS)
```

Comment:
here as well, the get_best_level_for_downsample  is likely not ideal, we can do better with the modifications you made on line 188